diff --git a/test/runner.rb b/test/runner.rb
index 3e2e1316b1..18111e45ab 100644
--- a/test/runner.rb
+++ b/test/runner.rb
@@ -4,7 +4,10 @@
 ENV["GEM_SKIP"] = ENV["GEM_HOME"] = ENV["GEM_PATH"] = "".freeze
 
 # Get bundled gems on load path
-Dir.glob("#{__dir__}/../gems/*/*.gemspec")
+
+gems = "#{RbConfig::CONFIG['rubylibdir'].sub('/ruby/', '/ruby/gems/')}/gems"
+
+Dir.glob("#{gems}/*/*.gemspec")
   .reject {|f| f =~ /minitest|test-unit|power_assert/ }
   .map {|f| $LOAD_PATH.unshift File.join(File.dirname(f), "lib") }
 
