Change asserts to use install folder. not build folder
diff --git a/test/irb/test_cmd.rb b/test/irb/test_cmd.rb
index 719a2bbf8f..ffa10bcb73 100644
--- a/test/irb/test_cmd.rb
+++ b/test/irb/test_cmd.rb
@@ -664,7 +664,7 @@ def test_edit_with_constant
         )
 
         assert_empty err
-        assert_match(/path: .*\/lib\/irb\.rb/, out)
+        assert_includes out, "path: #{RbConfig::CONFIG['rubylibdir']}/irb.rb"
         assert_match("command: ': code'", out)
       end
 
@@ -674,7 +674,7 @@ def test_edit_with_class_method
         )
 
         assert_empty err
-        assert_match(/path: .*\/lib\/irb\.rb/, out)
+        assert_includes out, "path: #{RbConfig::CONFIG['rubylibdir']}/irb.rb"
         assert_match("command: ': code'", out)
       end
 
@@ -684,7 +684,7 @@ def test_edit_with_instance_method
         )
 
         assert_empty err
-        assert_match(/path: .*\/lib\/irb\.rb/, out)
+        assert_includes out, "path: #{RbConfig::CONFIG['rubylibdir']}/irb.rb"
         assert_match("command: ': code'", out)
       end
     end
