diff --git a/test/irb/test_cmd.rb b/test/irb/test_cmd.rb
index 345f04bcf2..3655400ce1 100644
--- a/test/irb/test_cmd.rb
+++ b/test/irb/test_cmd.rb
@@ -802,6 +802,10 @@ def test_show_doc_without_rdoc
   end
 
   class EditTest < CommandTestCase
+
+    # used for running tests from install folder
+    LIB_PATH = RbConfig::CONFIG['rubylibdir']
+
     def setup
       @original_visual = ENV["VISUAL"]
       @original_editor = ENV["EDITOR"]
@@ -851,7 +855,7 @@ def test_edit_with_constant
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
@@ -861,7 +865,7 @@ def test_edit_with_class_method
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
@@ -871,7 +875,7 @@ def test_edit_with_instance_method
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
