diff --git a/test/irb/test_cmd.rb b/test/irb/test_cmd.rb
index 66328b89b1..8859c91bd9 100644
--- a/test/irb/test_cmd.rb
+++ b/test/irb/test_cmd.rb
@@ -7,6 +7,10 @@
 
 module TestIRB
   class CommandTestCase < TestCase
+  
+    # used for running tests from install folder
+    LIB_PATH = RbConfig::CONFIG['rubylibdir']
+      
     def setup
       @pwd = Dir.pwd
       @tmpdir = File.join(Dir.tmpdir, "test_reline_config_#{$$}")
@@ -860,7 +864,7 @@ def test_edit_with_constant
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
@@ -870,7 +874,7 @@ def test_edit_with_class_method
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
@@ -880,7 +884,7 @@ def test_edit_with_instance_method
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
   end
