diff --git a/test/irb/test_cmd.rb b/test/irb/test_cmd.rb
index bc63587330..d09351296f 100644
--- a/test/irb/test_cmd.rb
+++ b/test/irb/test_cmd.rb
@@ -824,6 +824,10 @@ def test_show_doc_without_rdoc
   end
 
   class EditTest < CommandTestCase
+
+    # used for running tests from install folder
+    LIB_PATH = RbConfig::CONFIG['rubylibdir']
+
     def setup
       @original_visual = ENV["VISUAL"]
       @original_editor = ENV["EDITOR"]
@@ -883,7 +887,7 @@ def test_edit_with_constant
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
@@ -893,7 +897,7 @@ def test_edit_with_class_method
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
@@ -903,7 +907,7 @@ def test_edit_with_instance_method
       )
 
       assert_empty err
-      assert_match(/path: .*\/lib\/irb\.rb/, out)
+      assert_include out, "path: #{LIB_PATH}/irb.rb"
       assert_match("command: ': code'", out)
     end
 
