diff --git a/test/json/test_helper.rb b/test/json/test_helper.rb
index e8bba16f84..93763fa96f 100644
--- a/test/json/test_helper.rb
+++ b/test/json/test_helper.rb
@@ -1,12 +1,12 @@
 case ENV['JSON']
 when 'pure'
-  $LOAD_PATH.unshift(File.expand_path('../../../lib', __FILE__))
+  # $LOAD_PATH.unshift(File.expand_path('../../../lib', __FILE__))
   require 'json/pure'
 when 'ext'
-  $LOAD_PATH.unshift(File.expand_path('../../../ext', __FILE__), File.expand_path('../../../lib', __FILE__))
+  # $LOAD_PATH.unshift(File.expand_path('../../../ext', __FILE__), File.expand_path('../../../lib', __FILE__))
   require 'json/ext'
 else
-  $LOAD_PATH.unshift(File.expand_path('../../../ext', __FILE__), File.expand_path('../../../lib', __FILE__))
+  # $LOAD_PATH.unshift(File.expand_path('../../../ext', __FILE__), File.expand_path('../../../lib', __FILE__))
   require 'json'
 end
 
