Patch by MSP-Greg
lower timeout time
diff --git a/test/lib/jit_support.rb b/test/lib/jit_support.rb
index b44c94f5eb..03a127f0f0 100644
--- a/test/lib/jit_support.rb
+++ b/test/lib/jit_support.rb
@@ -1,7 +1,7 @@
 require 'rbconfig'
 
 module JITSupport
-  JIT_TIMEOUT = 600 # 10min for each...
+  JIT_TIMEOUT = 60 # 1 min for each...
   JIT_SUCCESS_PREFIX = 'JIT success \(\d+\.\dms\)'
   JIT_RECOMPILE_PREFIX = 'JIT recompile'
   JIT_COMPACTION_PREFIX = 'JIT compaction \(\d+\.\dms\)'
