diff --git a/test/mkmf/base.rb b/test/mkmf/base.rb
index ec42bca100..192a6f7dd3 100644
--- a/test/mkmf/base.rb
+++ b/test/mkmf/base.rb
@@ -1,14 +1,12 @@
 # frozen_string_literal: false
-$extmk = true
-require 'rbconfig'
-RbConfig.fire_update!("top_srcdir", File.expand_path("../..", __dir__))
-File.foreach(RbConfig::CONFIG["topdir"]+"/Makefile") do |line|
-  if /^CC_WRAPPER\s*=\s*/ =~ line
-    RbConfig.fire_update!('CC_WRAPPER', $'.strip)
-    break
-  end
+
+begin
+  require 'devkit'
+rescue LoadError
 end
 
+$extmk = true
+require 'rbconfig'
 require 'test/unit'
 require 'mkmf'
 require 'tmpdir'
