diff --git a/test/net/smtp/test_sslcontext.rb b/test/net/smtp/test_sslcontext.rb
index 80afc1338e..ad7a708868 100644
--- a/test/net/smtp/test_sslcontext.rb
+++ b/test/net/smtp/test_sslcontext.rb
@@ -39,7 +39,7 @@ def teardown
     end
 
     def start_smtpd(starttls)
-      @server_socket, @client_socket = UNIXSocket.pair
+      @server_socket, @client_socket = Socket.pair :INET, :STREAM, 0
       @starttls_executed = false
       @server_thread = Thread.new(@server_socket) do |s|
         s.puts "220 fakeserver\r\n"
@@ -125,4 +125,4 @@ def test_start_without_tls_hostname
     end
 
   end
-end unless /mswin|mingw/ =~ RUBY_PLATFORM
+end
diff --git a/test/net/smtp/test_starttls.rb b/test/net/smtp/test_starttls.rb
index 2c812aea67..bc2db64297 100644
--- a/test/net/smtp/test_starttls.rb
+++ b/test/net/smtp/test_starttls.rb
@@ -25,7 +25,7 @@ def teardown
     end
 
     def start_smtpd(starttls)
-      @server_socket, @client_socket = UNIXSocket.pair
+      @server_socket, @client_socket = Socket.pair :INET, :STREAM, 0
       @starttls_executed = false
       @server_thread = Thread.new(@server_socket) do |s|
         s.puts "220 fakeserver\r\n"
@@ -118,4 +118,4 @@ def test_enable_starttls_and_enable_starttls_auto
       assert_nothing_raised { smtp.enable_starttls_auto }
     end
   end
-end unless /mswin|mingw/ =~ RUBY_PLATFORM
+end
