diff --git a/test/objspace/test_objspace.rb b/test/objspace/test_objspace.rb
index 1392447e4f..f945e10fc0 100644
--- a/test/objspace/test_objspace.rb
+++ b/test/objspace/test_objspace.rb
@@ -141,6 +141,8 @@ def test_reachable_objects_from
   end
 
   def test_reachable_objects_during_iteration
+    pend 'Skipped on mingw, intermittent failure' if ENV.key?('CI') && RUBY_PLATFORM.include?('mingw')
+
     opts = %w[--disable-gem --disable=frozen-string-literal -robjspace]
     assert_separately opts, "#{<<-"begin;"}\n#{<<-'end;'}"
     begin;
