diff --git a/test/openssl/test_ssl.rb b/test/openssl/test_ssl.rb
index a7607da073..01a5b652b9 100644
--- a/test/openssl/test_ssl.rb
+++ b/test/openssl/test_ssl.rb
@@ -97,11 +97,28 @@ def test_socket_open_with_context
     }
   end
 
+  def port_in_use?(host, port)
+    TCPServer.open(host, port).close
+    false
+  rescue Errno::EADDRINUSE, Errno::EACCES
+    true
+  end
+
+  def get_random_port(host = '127.0.0.1')
+    # IANA suggests dynamic port for 49152 to 65535
+    # http://www.iana.org/assignments/port-numbers
+    begin
+      port = rand(50_000..65_535)
+    end while port_in_use? host, port
+    port
+  end
+  private :port_in_use?, :get_random_port
+
   def test_socket_open_with_local_address_port_context
     start_server { |port|
       begin
         # Guess a free port number
-        random_port = rand(49152..65535)
+        random_port = get_random_port
         ctx = OpenSSL::SSL::SSLContext.new
         ssl = OpenSSL::SSL::SSLSocket.open("127.0.0.1", port, "127.0.0.1", random_port, context: ctx)
         ssl.sync_close = true
diff --git a/test/socket/test_addrinfo.rb b/test/socket/test_addrinfo.rb
index bdf1f7649e..07c4127e66 100644
--- a/test/socket/test_addrinfo.rb
+++ b/test/socket/test_addrinfo.rb
@@ -372,10 +372,8 @@ def random_port
   end
 
   def errors_addrinuse
-    errs = [Errno::EADDRINUSE]
-    # MinGW fails with "Errno::EACCES: Permission denied - bind(2) for 0.0.0.0:49721"
-    errs << Errno::EACCES if /mingw/ =~ RUBY_PLATFORM
-    errs
+    RUBY_PLATFORM.match?(/mingw|mswin/) ?
+      [Errno::EADDRINUSE, Errno::EACCES] : [Errno::EADDRINUSE]
   end
 
   def test_connect_from
diff --git a/test/socket/test_socket.rb b/test/socket/test_socket.rb
index 2c945aa9b9..d48e5e2f75 100644
--- a/test/socket/test_socket.rb
+++ b/test/socket/test_socket.rb
@@ -172,10 +172,8 @@ def random_port
   end
 
   def errors_addrinuse
-    errs = [Errno::EADDRINUSE]
-    # MinGW fails with "Errno::EACCES: Permission denied - bind(2) for 0.0.0.0:49721"
-    errs << Errno::EACCES if /mingw/ =~ RUBY_PLATFORM
-    errs
+    RUBY_PLATFORM.match?(/mingw|mswin/) ?
+      [Errno::EADDRINUSE, Errno::EACCES] : [Errno::EADDRINUSE]
   end
 
   def test_tcp_server_sockets
