# without this, intermittent failures in CI
diff --git a/test/rdoc/support/test_case.rb b/test/rdoc/support/test_case.rb
index d98dbe0d7b..6cdde617e9 100644
--- a/test/rdoc/support/test_case.rb
+++ b/test/rdoc/support/test_case.rb
@@ -13,7 +13,7 @@
 require 'tmpdir'
 require 'stringio'
 
-require_relative '../../../lib/rdoc'
+require 'rdoc'
 
 ##
 # RDoc::TestCase is an abstract TestCase to provide common setup and teardown
diff --git a/test/rdoc/test_rdoc_generator_json_index.rb b/test/rdoc/test_rdoc_generator_json_index.rb
index 62d1ccec95..58925a8424 100644
--- a/test/rdoc/test_rdoc_generator_json_index.rb
+++ b/test/rdoc/test_rdoc_generator_json_index.rb
@@ -95,7 +95,8 @@ def test_generate
     assert_file 'js/navigation.js'
     assert_file 'js/search_index.js'
 
-    srcdir = File.expand_path('lib/rdoc', @pwd)
+    #srcdir = File.expand_path('lib/rdoc', @pwd)
+    srcdir = "#{RbConfig::CONFIG['rubylibdir']}/rdoc"
     if !File.directory? srcdir
       # for Ruby core repository
       srcdir = File.expand_path("../../../lib/rdoc", __FILE__)
diff --git a/test/rdoc/test_rdoc_markdown.rb b/test/rdoc/test_rdoc_markdown.rb
index 72587caf46..ff508e57ee 100644
--- a/test/rdoc/test_rdoc_markdown.rb
+++ b/test/rdoc/test_rdoc_markdown.rb
@@ -2,8 +2,8 @@
 # frozen_string_literal: true
 
 require_relative 'helper'
-require_relative '../../lib/rdoc/markup/block_quote'
-require_relative '../../lib/rdoc/markdown'
+require 'rdoc/markup/block_quote'
+require 'rdoc/markdown'
 
 class TestRDocMarkdown < RDoc::TestCase
 
diff --git a/test/rdoc/test_rdoc_markdown_test.rb b/test/rdoc/test_rdoc_markdown_test.rb
index 193d648253..4def1a101f 100644
--- a/test/rdoc/test_rdoc_markdown_test.rb
+++ b/test/rdoc/test_rdoc_markdown_test.rb
@@ -2,8 +2,8 @@
 require_relative 'helper'
 require 'pp'
 
-require_relative '../../lib/rdoc'
-require_relative '../../lib/rdoc/markdown'
+require 'rdoc'
+require 'rdoc/markdown'
 
 class TestRDocMarkdownTest < RDoc::TestCase
 
diff --git a/test/rdoc/test_rdoc_rubygems_hook.rb b/test/rdoc/test_rdoc_rubygems_hook.rb
index 59a7ed0f89..9591ec42ef 100644
--- a/test/rdoc/test_rdoc_rubygems_hook.rb
+++ b/test/rdoc/test_rdoc_rubygems_hook.rb
@@ -2,7 +2,7 @@
 require 'rubygems'
 require 'fileutils'
 require 'tmpdir'
-require_relative '../../lib/rdoc/rubygems_hook'
+require 'rdoc/rubygems_hook'
 require 'test/unit'
 
 class TestRDocRubygemsHook < Test::Unit::TestCase
diff --git a/test/rdoc/test_rdoc_servlet.rb b/test/rdoc/test_rdoc_servlet.rb
index 7a5b15a6eb..965912c9d5 100644
--- a/test/rdoc/test_rdoc_servlet.rb
+++ b/test/rdoc/test_rdoc_servlet.rb
@@ -5,6 +5,9 @@
 rescue LoadError
 end
 
+require 'tsort' rescue nil
+require 'json'  rescue nil
+
 class TestRDocServlet < RDoc::TestCase
 
   def setup
