diff --git a/test/rubygems/bundler_test_gem.rb b/test/rubygems/bundler_test_gem.rb
index ca2980e04b..d730c74081 100644
--- a/test/rubygems/bundler_test_gem.rb
+++ b/test/rubygems/bundler_test_gem.rb
@@ -409,6 +409,8 @@ def with_local_bundler_at(path)
 
     # If bundler gemspec exists, pretend it's installed
     bundler_gemspec = File.expand_path("../../bundler/bundler.gemspec", __dir__)
+    bundler_gemspec = File.expand_path("../../lib/bundler/bundler.gemspec", __dir__) unless File.exist?(bundler_gemspec)
+
     if File.exist?(bundler_gemspec)
       target_gemspec_location = "#{path}/specifications/bundler-#{Bundler::VERSION}.gemspec"
 
diff --git a/test/rubygems/test_gem.rb b/test/rubygems/test_gem.rb
index cdc3479e37..c6f53b07cd 100644
--- a/test/rubygems/test_gem.rb
+++ b/test/rubygems/test_gem.rb
@@ -947,7 +947,7 @@ def test_self_platforms
   end
 
   def test_self_prefix
-    assert_equal PROJECT_DIR, Gem.prefix
+    assert_nil Gem.prefix
   end
 
   def test_self_prefix_libdir
diff --git a/test/rubygems/test_gem_commands_environment_command.rb b/test/rubygems/test_gem_commands_environment_command.rb
index 48252d84d4..e4f130936e 100644
--- a/test/rubygems/test_gem_commands_environment_command.rb
+++ b/test/rubygems/test_gem_commands_environment_command.rb
@@ -29,7 +29,13 @@ def test_execute
                  @ui.output)
     assert_match(/USER INSTALLATION DIRECTORY: #{Regexp.escape Gem.user_dir}/,
                  @ui.output)
-    assert_match(/RUBYGEMS PREFIX: /, @ui.output)
+
+    if Gem.prefix.nil?
+      refute_match(/RUBYGEMS PREFIX: /, @ui.output)
+    else
+      assert_match(/RUBYGEMS PREFIX: /, @ui.output)
+    end
+
     assert_match(/RUBY EXECUTABLE:.*#{RbConfig::CONFIG["ruby_install_name"]}/,
                  @ui.output)
     assert_match(/GIT EXECUTABLE: #{@cmd.send(:git_path)}/, @ui.output)
diff --git a/test/rubygems/test_gem_commands_update_command.rb b/test/rubygems/test_gem_commands_update_command.rb
index 3b106e4581..8774df1d9c 100644
--- a/test/rubygems/test_gem_commands_update_command.rb
+++ b/test/rubygems/test_gem_commands_update_command.rb
@@ -761,8 +761,11 @@ def test_update_rubygems_arguments
 
     arguments = @cmd.update_rubygems_arguments
 
-    assert_equal "--prefix",           arguments.shift
-    assert_equal Gem.prefix,           arguments.shift
+    # asserts fail from install folder
+    if Gem.method(:host).source_location[0].start_with? "#{File.expand_path "../../..", __FILE__}/"
+      assert_equal "--prefix",           arguments.shift
+      assert_equal Gem.prefix,           arguments.shift
+    end
     assert_equal "--no-document",      arguments.shift
     assert_equal "--previous-version", arguments.shift
     assert_equal Gem::VERSION,         arguments.shift
diff --git a/test/rubygems/test_gem_ext_builder.rb b/test/rubygems/test_gem_ext_builder.rb
index 34f85e6b75..e45ab6544c 100644
--- a/test/rubygems/test_gem_ext_builder.rb
+++ b/test/rubygems/test_gem_ext_builder.rb
@@ -247,7 +247,7 @@ def test_build_extensions_extconf_bad
     gem_make_out = File.join @spec.extension_dir, "gem_make.out"
     cmd_make_out = File.read(gem_make_out)
 
-    assert_match %r{#{Regexp.escape Gem.ruby} .* extconf\.rb}, cmd_make_out
+    assert_match %r{/bin/ruby.exe( .*)? extconf\.rb}, cmd_make_out
     assert_match(/: No such file/, cmd_make_out)
 
     assert_path_not_exist @spec.gem_build_complete_path
diff --git a/test/rubygems/test_gem_ext_cargo_builder.rb b/test/rubygems/test_gem_ext_cargo_builder.rb
index b970e442c2..e97bcf10be 100644
--- a/test/rubygems/test_gem_ext_cargo_builder.rb
+++ b/test/rubygems/test_gem_ext_cargo_builder.rb
@@ -201,7 +201,8 @@ def skip_unsupported_platforms!
     system(@rust_envs, "cargo", "-V", out: IO::NULL, err: [:child, :out])
     pend "cargo not present" unless $?.success?
     pend "ruby.h is not provided by ruby repo" if ruby_repo?
-    pend "rust toolchain of mingw is broken" if mingw_windows?
+    # ruby-loco pend "rust toolchain of mingw is broken" if mingw_windows?
+    pend "rust toolchains of mingw and msvc are broken" if mingw_windows? || vc_windows?
   end
 
   def assert_ffi_handle(bundle, name)
diff --git a/test/rubygems/test_require.rb b/test/rubygems/test_require.rb
index f63c23c315..58c2326cc0 100644
--- a/test/rubygems/test_require.rb
+++ b/test/rubygems/test_require.rb
@@ -466,7 +466,7 @@ def test_realworld_default_gem
       require "json"
       puts Gem.loaded_specs["json"]
     RUBY
-    output = Gem::Util.popen(*ruby_with_rubygems_in_load_path, "-e", cmd).strip
+    output = Gem::Util.popen("ruby", "-e", cmd).strip
     assert $?.success?
     refute_empty output
   end
@@ -486,7 +486,8 @@ def test_realworld_upgraded_default_gem
     RUBY
     File.write(path, code)
 
-    output = Gem::Util.popen({ "GEM_HOME" => @gemhome }, *ruby_with_rubygems_in_load_path, path).strip
+    #output = Gem::Util.popen({ "GEM_HOME" => @gemhome }, *ruby_with_rubygems_in_load_path, path).strip
+    output = Gem::Util.popen({ "GEM_HOME" => @gemhome }, "ruby", path).strip
     refute_empty output
     assert_equal "999.99.9", output.lines[0].chomp
     # Make sure only files from the newer json gem are loaded, and no files from the default json gem
