# extend timeout to pass Windows parallel CI
diff --git a/test/ruby/test_file_exhaustive.rb b/test/ruby/test_file_exhaustive.rb
index 0ccf99312b..d4ebddbf23 100644
--- a/test/ruby/test_file_exhaustive.rb
+++ b/test/ruby/test_file_exhaustive.rb
@@ -1396,7 +1396,8 @@ def test_flock_exclusive
       assert_separately(["-rtimeout", "-", regular_file], "#{<<~begin}#{<<~"end;"}")
       begin
         open(ARGV[0], "r") do |f|
-          Timeout.timeout(0.1) do
+          t_o = RUBY_PLATFORM.match?(/mingw|mswin/) ? 0.3 : 0.1
+          Timeout.timeout(t_o) do
             assert(!f.flock(File::LOCK_SH|File::LOCK_NB))
           end
         end
@@ -1405,7 +1406,8 @@ def test_flock_exclusive
       begin
         open(ARGV[0], "r") do |f|
           assert_raise(Timeout::Error) do
-            Timeout.timeout(0.1) do
+            t_o = RUBY_PLATFORM.match?(/mingw|mswin/) ? 0.3 : 0.1
+            Timeout.timeout(t_o) do
               f.flock(File::LOCK_SH)
             end
           end
