diff --git a/test/ruby/test_exception.rb b/test/ruby/test_exception.rb
index 85ca4dd34a..442a36b2fd 100644
--- a/test/ruby/test_exception.rb
+++ b/test/ruby/test_exception.rb
@@ -995,7 +995,7 @@ def test_nomethod_error_new_receiver
     error = NoMethodError.new("Message", :foo)
     assert_raise(ArgumentError) {error.receiver}
 
-    msg = defined?(:DidYouMean.formatter) ?
+    msg = defined?(DidYouMean.formatter) ?
       "Message\nDid you mean?  for" : "Message"
 
     error = NoMethodError.new("Message", :foo, receiver: receiver)
