diff --git a/test/ruby/test_ast.rb b/test/ruby/test_ast.rb
index c8617d50f0..6c22fcab62 100644
--- a/test/ruby/test_ast.rb
+++ b/test/ruby/test_ast.rb
@@ -115,24 +115,22 @@ def validate_not_cared0(node)
   SRCDIR = File.expand_path("../../..", __FILE__)
 
   Dir.glob("test/**/*.rb", base: SRCDIR).each do |path|
+    next if path.start_with? 'test/tmp/'
+
     define_method("test_ranges:#{path}") do
       helper = Helper.new("#{SRCDIR}/#{path}")
       helper.validate_range
 
       assert_equal([], helper.errors)
     end
-  end
 
-  Dir.glob("test/**/*.rb", base: SRCDIR).each do |path|
     define_method("test_not_cared:#{path}") do
       helper = Helper.new("#{SRCDIR}/#{path}")
       helper.validate_not_cared
 
       assert_equal([], helper.errors)
     end
-  end
 
-  Dir.glob("test/**/*.rb", base: SRCDIR).each do |path|
     define_method("test_all_tokens:#{path}") do
       node = RubyVM::AbstractSyntaxTree.parse_file("#{SRCDIR}/#{path}", keep_tokens: true)
       tokens = node.all_tokens.sort_by { [_1.last[0], _1.last[1]] }
