diff --git a/test/ruby/test_process.rb b/test/ruby/test_process.rb
index 1d04d22631..38a51ef94f 100644
--- a/test/ruby/test_process.rb
+++ b/test/ruby/test_process.rb
@@ -23,6 +23,10 @@ def self.windows?
     return /mswin|mingw|bccwin/ =~ RUBY_PLATFORM
   end
 
+  def mswin?
+    return RUBY_PLATFORM.include?("mswin")
+  end
+
   def write_file(filename, content)
     File.open(filename, "w") {|f|
       f << content
@@ -320,7 +324,11 @@ def test_execopts_env
     end
     cmd << '-e' << 'puts ENV.keys.map{|e|e.upcase}'
     IO.popen(cmd) {|io|
-      assert_equal("PATH\n", io.read)
+      if mswin?
+      assert_includes(io.read, "PATH\n")
+      else
+        assert_equal("PATH\n", io.read)
+      end
     }
 
     IO.popen([{"FOO"=>"BAR"}, *ENVCOMMAND]) {|io|
@@ -444,10 +452,19 @@ def test_execopts_unsetenv_others
     h = {}
     MANDATORY_ENVS.each {|k| e = ENV[k] and h[k] = e}
     IO.popen([h, *ENVCOMMAND, :unsetenv_others=>true]) {|io|
-      assert_equal("", io.read)
+      if mswin?
+        str = io.read.lines.reject { |l| l.include? "SSL_"}.join('').strip
+        assert_equal("", str)
+      else
+        assert_equal("", io.read)
+      end
     }
     IO.popen([h.merge("A"=>"B"), *ENVCOMMAND, :unsetenv_others=>true]) {|io|
-      assert_equal("A=B\n", io.read)
+      if mswin?
+        assert_includes(io.read, "A=B\n")
+      else
+        assert_equal("A=B\n", io.read)
+      end
     }
   end
 
