diff --git a/test/runner.rb b/test/runner.rb
index 2f3267f6c3..786710da3c 100644
--- a/test/runner.rb
+++ b/test/runner.rb
@@ -6,11 +6,4 @@
 
 Warning[:experimental] = false
 
-gem_path = [
-  File.realdirpath(".bundle"),
-  File.realdirpath("../.bundle", __dir__),
-]
-ENV["GEM_PATH"] = gem_path.join(File::PATH_SEPARATOR)
-ENV["GEM_HOME"] = gem_path.first
-
 require_relative '../tool/test/runner'
