diff --git a/test/socket/test_addrinfo.rb b/test/socket/test_addrinfo.rb
index 92b61ea99e..ab8e08e8cd 100644
--- a/test/socket/test_addrinfo.rb
+++ b/test/socket/test_addrinfo.rb
@@ -367,7 +367,11 @@ def random_port
   end
 
   def errors_addrinuse
-    [Errno::EADDRINUSE]
+    if ENV['CI'] && RUBY_PLATFORM.match?(/mingw|mswin/)
+      [Errno::EADDRINUSE, Errno::EACCES]
+    else
+      [Errno::EADDRINUSE]
+    end
   end
 
   def test_connect_from
