diff --git a/test/socket/test_addrinfo.rb b/test/socket/test_addrinfo.rb
index d516c772d9..8ea3220c09 100644
--- a/test/socket/test_addrinfo.rb
+++ b/test/socket/test_addrinfo.rb
@@ -367,10 +367,12 @@ def random_port
   end
 
   def errors_addrinuse
-    errs = [Errno::EADDRINUSE]
-    # MinGW fails with "Errno::EACCES: Permission denied - bind(2) for 0.0.0.0:49721"
-    errs << Errno::EACCES if /mingw/ =~ RUBY_PLATFORM
-    errs
+    # Actions Window fails with "Errno::EACCES: Permission denied - bind(2) for 0.0.0.0:49721"
+    if ENV['CI'] && RUBY_PLATFORM.match?(/mingw|mswin/)
+      [Errno::EADDRINUSE, Errno::EACCES]
+    else
+      [Errno::EADDRINUSE]
+    end
   end
 
   def test_connect_from
