diff --git a/tool/test/init.rb b/tool/test/init.rb
index 3fd1419a9c..e780a7ebb9 100644
--- a/tool/test/init.rb
+++ b/tool/test/init.rb
@@ -6,9 +6,9 @@
   # Preserve the gem environment prepared by tool/runruby.rb for test-tool, which uses bundled gems.
   ENV["BUNDLED_#{gem_env}"] = ENV[gem_env]
 
-  ENV[gem_env] = "".freeze
+  # ENV[gem_env] = "".freeze
 end
-ENV["GEM_SKIP"] = "".freeze
+# ENV["GEM_SKIP"] = "".freeze
 
 ENV.delete("RUBY_CODESIGN")
 
