diff --git a/tool/lib/core_assertions.rb b/tool/lib/core_assertions.rb
index cc4eb327d4..a97875fa1e 100644
--- a/tool/lib/core_assertions.rb
+++ b/tool/lib/core_assertions.rb
@@ -820,18 +820,22 @@ def assert_all_assertions_foreach(msg = nil, *keys, &block)
       end
       alias all_assertions_foreach assert_all_assertions_foreach
 
-      %w[
-        CLOCK_THREAD_CPUTIME_ID CLOCK_PROCESS_CPUTIME_ID
-        CLOCK_MONOTONIC
-      ].find do |c|
-        if Process.const_defined?(c)
-          [c.to_sym, Process.const_get(c)].find do |clk|
-            begin
-              Process.clock_gettime(clk)
-            rescue
-              # Constants may be defined but not implemented, e.g., mingw.
-            else
-              PERFORMANCE_CLOCK = clk
+      if RUBY_PLATFORM.match?(/mswin|mingw/)
+        PERFORMANCE_CLOCK = Process::CLOCK_MONOTONIC
+      else
+        %w[
+          CLOCK_THREAD_CPUTIME_ID CLOCK_PROCESS_CPUTIME_ID
+          CLOCK_MONOTONIC
+        ].find do |c|
+          if Process.const_defined?(c) && !Process.const_get(c).nil?
+            [c.to_sym, Process.const_get(c)].find do |clk|
+              begin
+                Process.clock_gettime(clk)
+              rescue
+                # Constants may be defined but not implemented, e.g., mingw.
+              else
+                PERFORMANCE_CLOCK = clk
+              end
             end
           end
         end
