diff --git a/tool/test/init.rb b/tool/test/init.rb
index 3a1143d01d..2187fd4386 100644
--- a/tool/test/init.rb
+++ b/tool/test/init.rb
@@ -1,7 +1,7 @@
 # This file includes the settings for "make test-all".
 # Note that this file is loaded not only by test/runner.rb but also by tool/lib/test/unit/parallel.rb.
 
-ENV["GEM_SKIP"] = ENV["GEM_HOME"] = ENV["GEM_PATH"] = "".freeze
+# ENV["GEM_SKIP"] = ENV["GEM_HOME"] = ENV["GEM_PATH"] = "".freeze
 ENV.delete("RUBY_CODESIGN")
 
 Warning[:experimental] = false
