diff --git a/lib/ruby_installer/build/ca_cert_file.rb b/lib/ruby_installer/build/ca_cert_file.rb
index dd9009a..abe3004 100644
--- a/lib/ruby_installer/build/ca_cert_file.rb
+++ b/lib/ruby_installer/build/ca_cert_file.rb
@@ -17,7 +17,7 @@ class CaCertFile
     require 'openssl'
     require 'stringio'
 
-    csv_data = OpenURI.open_uri(MOZILLA_CA_CSV_URI)
+    csv_data = OpenURI.open_uri(MOZILLA_CA_CSV_URI, :ssl_ca_cert => OpenSSL::X509::DEFAULT_CERT_FILE)
 
     fd = StringIO.new
     fd.write <<-EOT
