From 31a3982c6a6f20e6520b601ef0f074bfcbb3053d Mon Sep 17 00:00:00 2001
From: Samuel Williams <samuel.williams@oriontransfer.co.nz>
Date: Wed, 21 Nov 2018 12:59:27 +1300
Subject: [PATCH] Use VirtualAlloc for windows stack allocation.

---
 cont.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cont.c b/cont.c
index 56128a843a71..ad4c99a65090 100644
--- a/cont.c
+++ b/cont.c
@@ -885,7 +885,7 @@ fiber_machine_stack_alloc(size_t size)
     }
     else {
 #ifdef _WIN32
-        return malloc(size);
+        return VirtualAlloc(0, size, MEM_COMMIT, PAGE_READWRITE);
 #else
 	void *page;
 	STACK_GROW_DIR_DETECTION;
