From ca41511de81dba9c848802973146e2dd6fcfee0f Mon Sep 17 00:00:00 2001
From: Samuel Williams <samuel.williams@oriontransfer.co.nz>
Date: Wed, 21 Nov 2018 14:36:25 +1300
Subject: [PATCH] Fix another stack release issue.

---
 cont.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cont.c b/cont.c
index 399e0846a73c..400a0f08bdbf 100644
--- a/cont.c
+++ b/cont.c
@@ -416,10 +416,11 @@ cont_free(void *ptr)
 		rb_bug("Illegal root fiber parameter");
 	    }
 #ifdef _WIN32
-            free((void*)fib->ss_sp);
+            VirtualFree((void*)fib->ss_sp, 0, MEM_RELEASE);
 #else
 	    munmap((void*)fib->ss_sp, fib->ss_size);
 #endif
+            fib->ss_sp = NULL;
 	}
 #elif defined(_WIN32)
 	if (!fiber_is_root_p(fib)) {
