From 8b842068c9f60d1e0a9e376c08d94f5ed119a7b9 Mon Sep 17 00:00:00 2001
From: Samuel Williams <samuel.williams@oriontransfer.co.nz>
Date: Wed, 21 Nov 2018 13:17:16 +1300
Subject: [PATCH] Must provide some pointer to VirtualProtect according to
 documentation.

---
 cont.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cont.c b/cont.c
index 778db5903ac7..7b75516caac4 100644
--- a/cont.c
+++ b/cont.c
@@ -870,6 +870,9 @@ static char*
 fiber_machine_stack_alloc(size_t size)
 {
     char *ptr;
+#ifdef _WIN32
+    DWORD old_protect;
+#endif
 
     if (machine_stack_cache_index > 0) {
         if (machine_stack_cache[machine_stack_cache_index - 1].size == (size / sizeof(VALUE))) {
@@ -888,7 +891,7 @@ fiber_machine_stack_alloc(size_t size)
             rb_raise(rb_eFiberError, "can't allocate machine stack to fiber: %s", ERRNOMSG);
         }
         
-        if (!VirtualProtect(ptr, RB_PAGE_SIZE, PAGE_READWRITE | PAGE_GUARD, NULL)) {
+        if (!VirtualProtect(ptr, RB_PAGE_SIZE, PAGE_READWRITE | PAGE_GUARD, &old_protect)) {
             rb_raise(rb_eFiberError, "can't set a guard page: %s", ERRNOMSG);
         }
 #else
