Patch by MSP-Greg
index 1d19af1e10..fb87085f87 100644
--- a/test/readline/test_readline.rb
+++ b/test/readline/test_readline.rb
@@ -412,18 +412,21 @@ def test_modify_text_in_pre_input_hook
   end if !/EditLine|\A4\.3\z/n.match(Readline::VERSION)
 
   def test_input_metachar
-    bug6601 = '[ruby-core:45682]'
-    Readline::HISTORY << "hello"
-    wo = nil
-    line = with_pipe do |r, w|
-      wo = w.dup
-      wo.write("\C-re\ef\n")
+    skip("Won't pass on Windows w/readline 7.0.003 [ruby-core:45682]") if windows?
+    begin
+      bug6601 = '[ruby-core:45682]'
+      Readline::HISTORY << "hello"
+      wo = nil
+      line = with_pipe do |r, w|
+        wo = w.dup
+        wo.write("\C-re\ef\n")
+      end
+      assert_equal("hello", line, bug6601)
+    ensure
+      wo.close
+      Readline.delete_text
+      Readline::HISTORY.clear
     end
-    assert_equal("hello", line, bug6601)
-  ensure
-    wo.close
-    Readline.delete_text
-    Readline::HISTORY.clear
   end if !/EditLine/n.match(Readline::VERSION)
 
   def test_input_metachar_multibyte
@@ -580,11 +583,15 @@ def replace_stdio(stdin_path, stdout_path)
   end
 
   def with_temp_stdio
-    Tempfile.create("test_readline_stdin") {|stdin|
-      Tempfile.create("test_readline_stdout") {|stdout|
-        yield stdin, stdout
-      }
-    }
+    stdin =  Tempfile.new("test_readline_stdin")
+    stdout = Tempfile.new("test_readline_stdout")
+    yield stdin, stdout
+    if stdout
+      stdout.closed? ? stdout.unlink : stdout.close!
+    end
+    if stdin
+      stdin.closed?  ? stdin.unlink  : stdin.close!
+    end
   end
 
   def with_pipe
