diff --git a/tool/downloader.rb b/tool/downloader.rb
index d09afc5ad7..72730800aa 100644
--- a/tool/downloader.rb
+++ b/tool/downloader.rb
@@ -76,7 +76,7 @@ def self.download(name, dir = nil, since = true, options = {})
 
   class Unicode < self
     INDEX = {}  # cache index file information across files in the same directory
-    UNICODE_PUBLIC = "https://www.unicode.org/Public/"
+    UNICODE_PUBLIC = "http://www.unicode.org/Public/"
 
     def self.download(name, dir = nil, since = true, options = {})
       options = options.dup
