diff --git a/tool/downloader.rb b/tool/downloader.rb
index bf5498f42d..12195e992f 100644
--- a/tool/downloader.rb
+++ b/tool/downloader.rb
@@ -71,7 +71,7 @@ def self.download(name, dir = nil, since = true, options = {})
 
   class Unicode < self
     def self.download(name, *rest)
-      super("http://www.unicode.org/Public/#{name}", name, *rest)
+      super("https://unicode.org/Public/#{name}", name, *rest)
     end
   end
 
