diff --git a/config.h b/config_b.h
index 4c97795..151908e 100644
--- a/.ext/include/x64-mingw32/ruby/config.h
+++ b/.ext/include/x64-mingw32/ruby/config.h
@@ -288,8 +288,8 @@
 #define LIBDIR_BASENAME "lib"
 #define EXECUTABLE_EXTS ".exe",".com",".cmd",".bat"
 #define HAVE__SETJMPEX 1
-#define RUBY_SETJMP(env) setjmpex((env))
-#define RUBY_LONGJMP(env,val) longjmp((env),val)
+#define RUBY_SETJMP(env) __builtin_setjmp((env))
+#define RUBY_LONGJMP(env,val) __builtin_longjmp((env),val)
 #define RUBY_JMP_BUF jmp_buf
 #define RUBY_USE_SETJMPEX 1
 #define USE_MJIT 1
