diff --git a/spec/ruby/library/net/http/http/fixtures/http_server.rb b/spec/ruby/library/net/http/http/fixtures/http_server.rb
index aeb0babda1..a84e0d47d0 100644
--- a/spec/ruby/library/net/http/http/fixtures/http_server.rb
+++ b/spec/ruby/library/net/http/http/fixtures/http_server.rb
@@ -64,7 +64,7 @@ def port

     def start_server
       server_config = {
-        BindAddress: "127.0.0.1",
+        BindAddress: "localhost",
         Port: 0,
         Logger: WEBrick::Log.new(NullWriter.new),
         AccessLog: [],
