diff --git a/ext/openssl/extconf.rb b/ext/openssl/extconf.rb
index 467f7e7a86..869f0b1f08 100644
--- a/ext/openssl/extconf.rb
+++ b/ext/openssl/extconf.rb
@@ -184,7 +184,7 @@ def find_openssl_library
 
 Logging::message "=== Checking done. ===\n"
 
-if openssl_3
+if openssl_3 && $warnflags
   if $warnflags.sub!(/-W\K(?=deprecated-declarations)/, 'no-')
     $warnflags << " -Wno-incompatible-pointer-types-discards-qualifiers"
   end
