diff --git a/gems/bundled_gems b/gems/bundled_gems
index 93049f93a6..920939fcaf 100644
--- a/gems/bundled_gems
+++ b/gems/bundled_gems
@@ -7,7 +7,7 @@
 #   if `revision` is not given, "v"+`version` or `version` will be used.
 
 minitest            5.25.5  https://github.com/minitest/minitest
-power_assert        2.0.5   https://github.com/ruby/power_assert a7dab941153b233d3412e249d25da52a6c5691de
+power_assert        2.0.6dev  https://github.com/ruby/power_assert a7dab941153b233d3412e249d25da52a6c5691de
 rake                13.3.0  https://github.com/ruby/rake
 test-unit           3.6.8   https://github.com/test-unit/test-unit
 rexml               3.4.1   https://github.com/ruby/rexml
