diff --git a/gems/bundled_gems b/gems/bundled_gems
index f70bdd6c5b..6bae198fc9 100644
--- a/gems/bundled_gems
+++ b/gems/bundled_gems
@@ -1,3 +1,4 @@
+
 # gem-name version repository-url [revision]
 #
 # - gem-name: gem name to bundle
@@ -33,7 +34,7 @@ rinda               0.2.0   https://github.com/ruby/rinda
 drb                 2.2.1   https://github.com/ruby/drb
 nkf                 0.2.0   https://github.com/ruby/nkf
 syslog              0.3.0   https://github.com/ruby/syslog
-csv                 3.3.3   https://github.com/ruby/csv e0874e47e2730dacccdb0c534a2cdc073d3bdaea
+csv                 3.3.4   https://github.com/ruby/csv e0874e47e2730dacccdb0c534a2cdc073d3bdaea
 repl_type_completor 0.1.10  https://github.com/ruby/repl_type_completor
 ostruct             0.6.1   https://github.com/ruby/ostruct
 pstore              0.2.0   https://github.com/ruby/pstore
