diff --git a/spec/ruby/core/kernel/require_spec.rb b/spec/ruby/core/kernel/require_spec.rb
index c609809df5..f057c71349 100644
--- a/spec/ruby/core/kernel/require_spec.rb
+++ b/spec/ruby/core/kernel/require_spec.rb
@@ -22,7 +22,11 @@
       provided += %w[set pathname]
     end
     ruby_version_is "4.1" do
-      provided += %w[monitor]
+      if RUBY_PLATFORM.include?('mswin')
+        provided += %w[monitor rbconfig]
+      else
+        provided += %w[monitor]
+      end
     end
 
     out = ruby_exe("puts $LOADED_FEATURES", options: '--disable-gems --disable-did-you-mean')
