diff --git a/spec/ruby/core/main/using_spec.rb b/spec/ruby/core/main/using_spec.rb
index 1fb812f5cc..34b507b520 100644
--- a/spec/ruby/core/main/using_spec.rb
+++ b/spec/ruby/core/main/using_spec.rb
@@ -1,15 +1,15 @@
 require_relative '../../spec_helper'
 require_relative 'fixtures/classes'
 
+require 'open3'
+
 describe "main.using" do
   it "requires one Module argument" do
-    lambda do
-      eval('using', TOPLEVEL_BINDING)
-    end.should raise_error(ArgumentError)
+    _, _, stderr, _ = Open3.popen3("ruby -e \"eval('using', TOPLEVEL_BINDING)\"")
+    stderr.read.should =~ /\(ArgumentError\)/
 
-    lambda do
-      eval('using "foo"', TOPLEVEL_BINDING)
-    end.should raise_error(TypeError)
+    _, _, stderr, _ = Open3.popen3("ruby -e \"eval('using :foo', TOPLEVEL_BINDING)\"")
+    stderr.read.should =~ /\(TypeError\)/
   end
 
   it "uses refinements from the given module only in the target file" do
