diff --git a/spec/ruby/core/kernel/require_spec.rb b/spec/ruby/core/kernel/require_spec.rb
index 896afb840a..70afb4d170 100644
--- a/spec/ruby/core/kernel/require_spec.rb
+++ b/spec/ruby/core/kernel/require_spec.rb
@@ -26,7 +26,11 @@
     features = out.lines.map { |line| File.basename(line.chomp, '.*') }
 
     # Ignore CRuby internals
-    features -= %w[encdb transdb windows_1252]
+    if RUBY_PLATFORM.include? 'mswin'
+      features -= %w[encdb transdb windows_1252 rbconfig]
+    else
+      features -= %w[encdb transdb windows_1252]
+    end
     features.reject! { |feature| feature.end_with?('-fake') }
 
     features.sort.should == provided.sort
