diff --git a/spec/ruby/core/kernel/require_spec.rb b/spec/ruby/core/kernel/require_spec.rb
index da2f48fb61..a86e84b677 100644
--- a/spec/ruby/core/kernel/require_spec.rb
+++ b/spec/ruby/core/kernel/require_spec.rb
@@ -23,6 +23,7 @@
   end
   ruby_version_is "4.1" do
     provided << "monitor"
+    provided << "rbconfig" if RUBY_PLATFORM.include?('mswin')
   end
 
   it "#{provided.join(', ')} are already required" do
