diff --git a/spec/ruby/core/file/join_spec.rb b/spec/ruby/core/file/join_spec.rb
index 0f0911ea31..f027cfc7b2 100644
--- a/spec/ruby/core/file/join_spec.rb
+++ b/spec/ruby/core/file/join_spec.rb
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
 require_relative '../../spec_helper'
 
 describe "File.join" do
@@ -66,11 +68,15 @@
 
   it "inserts the separator in between empty strings and arrays" do
     File.join("").should == ""
+    sleep 0.3
     File.join("", "").should == "/"
+    sleep 0.3
     File.join(["", ""]).should == "/"
+    sleep 0.3
     File.join("a", "").should == "a/"
+    sleep 0.3
     File.join("", "a").should == "/a"
-
+    sleep 0.3
     File.join([]).should == ""
     File.join([], []).should == "/"
     File.join([[], []]).should == "/"
@@ -79,9 +85,13 @@
   end
 
   it "handles leading parts edge cases" do
+    sleep 0.5
     File.join("/bin")     .should == "/bin"
+    sleep 0.5
     File.join("", "bin")  .should == "/bin"
+    sleep 0.5
     File.join("/", "bin") .should == "/bin"
+    sleep 0.5
     File.join("/", "/bin").should == "/bin"
   end
 
