diff --git a/win32/setup.mak b/win32/setup.mak
index 15fa41b562..ba3f66d7d3 100644
--- a/win32/setup.mak
+++ b/win32/setup.mak
@@ -22,7 +22,7 @@ MAKE = $(MAKE) -f $(MAKEFILE)
 MAKEFILE = Makefile
 !endif
 CPU = PROCESSOR_LEVEL
-CC = $(CC) -nologo
+CC = $(CC) -nologo $(XINCFLAGS)
 CPP = $(CC) -EP
 
 all: -prologue- -generic- -epilogue-
@@ -188,7 +188,10 @@ echo TEENY = RUBY_VERSION_TEENY
 echo ABI_VERSION = RUBY_ABI_VERSION
 #endif
 set /a MSC_VER = _MSC_VER
-#if _MSC_VER >= 1920
+#if _MSC_VER >= 1940
+set /a MSC_VER_LOWER = (MSC_VER-20)/20*20+0
+set /a MSC_VER_UPPER = MSC_VER/20*20+19
+#elif _MSC_VER >= 1920
 set /a MSC_VER_LOWER = MSC_VER/20*20+0
 set /a MSC_VER_UPPER = MSC_VER/20*20+19
 #elif _MSC_VER >= 1900
