Patch by MSP-Greg
diff --git a/include/ruby/win32.h b/include/ruby/win32.h
index 6db757b6e3..24d1639f61 100644
--- a/include/ruby/win32.h
+++ b/include/ruby/win32.h
@@ -194,7 +194,7 @@ struct stati128 {
 #if SIZEOF_OFF_T == 8
 #define off_t __int64
 #define stat stati128
-#define SIZEOF_STRUCT_STAT_ST_INO sizeof(unsigned __int64)
+/* #define SIZEOF_STRUCT_STAT_ST_INO sizeof(unsigned __int64) */
 #define HAVE_STRUCT_STAT_ST_INOHIGH
 #define HAVE_STRUCT_STAT_ST_ATIMENSEC
 #define HAVE_STRUCT_STAT_ST_MTIMENSEC
