diff --git a/lib/mkmf.rb b/lib/mkmf.rb
index 73459ffeb9..e9a92703e0 100644
--- a/lib/mkmf.rb
+++ b/lib/mkmf.rb
@@ -248,6 +248,10 @@ def map_dir(dir, map = nil)
   elsif File.exist?(($hdrdir = ($top_srcdir ||= topdir) + "/include")  + "/ruby.h")
     $topdir ||= RbConfig::CONFIG["topdir"]
     $arch_hdrdir = "$(extout)/include/$(arch)"
+  elsif File.exist?(($hdrdir = RbConfig::CONFIG["rubyhdrdir"]) + "/ruby/ruby.h")
+    $extmk = false
+    $topdir = $top_srcdir = $hdrdir
+    $arch_hdrdir = RbConfig::CONFIG["rubyarchhdrdir"]
   else
     abort <<MESSAGE
 mkmf.rb can't find header files for ruby at #{$hdrdir}/ruby.h
