diff --git a/bootstraptest/runner.rb b/bootstraptest/runner.rb
index 8d30d3743c..a6ea086f83 100755
--- a/bootstraptest/runner.rb
+++ b/bootstraptest/runner.rb
@@ -789,9 +789,15 @@ def assert_finish(timeout_seconds, testsrc, message = '')
         end
         if io.respond_to?(:read_nonblock)
           if IO.select([io], nil, nil, diff)
+            tries = 0
             begin
               io.read_nonblock(1024)
-            rescue Errno::EAGAIN, IO::WaitReadable, EOFError
+            rescue IO::WaitReadable
+              IO.select([io])
+              tries += 1
+              break if tries > 3
+              retry
+            rescue Errno::EAGAIN, EOFError
               break
             end while true
           end
diff --git a/bootstraptest/test_io.rb b/bootstraptest/test_io.rb
index 4e5d6d59c9..547dbc2305 100644
--- a/bootstraptest/test_io.rb
+++ b/bootstraptest/test_io.rb
@@ -1,5 +1,5 @@
 /freebsd/ =~ RUBY_PLATFORM or
-assert_finish 5, %q{
+assert_finish 6, %q{
   r, w = IO.pipe
   t1 = Thread.new { r.sysread(1) }
   t2 = Thread.new { r.sysread(1) }
diff --git a/bootstraptest/test_thread.rb b/bootstraptest/test_thread.rb
index 4040b68a27..87d09d5f64 100644
--- a/bootstraptest/test_thread.rb
+++ b/bootstraptest/test_thread.rb
@@ -347,7 +347,7 @@ def m
 
 assert_normal_exit %q{
   Thread.new("foo", &Object.method(:class_eval)).join
-}, '[ruby-dev:34128]'
+}, '[ruby-dev:34128]', 'SIGSEGV'
 
 assert_equal 'ok', %q{
   begin
