diff --git a/file.c b/file.c
index b24b671bb2..47e22453c1 100644
--- a/file.c
+++ b/file.c
@@ -2917,9 +2917,9 @@ rb_file_s_rename(VALUE klass, VALUE from, VALUE to)
 #if defined DOSISH
 	switch (e) {
 	  case EEXIST:
-	    if (chmod(dst, 0666) == 0 &&
-		unlink(dst) == 0 &&
-		rename(src, dst) == 0)
+	    if (chmod(ra.dst, 0666) == 0 &&
+		unlink(ra.dst) == 0 &&
+		rename(ra.src, ra.dst) == 0)
 		return INT2FIX(0);
 	}
 #endif
